code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/TimerPushTemplate.kt

```diff
@@ -0,0 +1,73 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.templates
+
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants.TimerKeys
+import com.adobe.marketing.mobile.notificationbuilder.internal.util.NotificationData
+
+/**
+ * Class for parsing the data required to display a Timer notification.
+ */
+internal class TimerPushTemplate(data: NotificationData) : AEPPushTemplate(data) {
+
+    internal var alternateTitle: String
+    internal var alternateBody: String?
+    internal var alternateExpandedBody: String?
+    internal var alternateImage: String?
+    internal var timerColor: String?
+    internal var expiryTime: Long
+    private val currentTimeInSeconds: Long
+        get() = System.currentTimeMillis() / 1000
+
+    val remainingTimeInSeconds: Long
+        get() = expiryTime - currentTimeInSeconds
+
+    /**
+     * Initialize the TimerPushTemplate with the provided [NotificationData].
+     */
+    init {
+        alternateTitle = data.getRequiredString(TimerKeys.ALTERNATE_TITLE)
```

Comment:
i see that alternate title is a required string. can we make sure to update the payload wiki with the latest required and optional values?

Reply:
Yup! This key was made mandatory during iOS template discussion. I will make sure to update the Wiki 👍🏼